src/QuadForm/indefiniteLLL.jl

```diff
@@ -0,0 +1,357 @@
+
+################################################################################
+#
+#                       indefinite LLL reduction
+#
+################################################################################
+
+################################################################################
+#                  Helpful
+################################################################################
+
+#=
+  _round_matrix(M::MatElem) -> MatElem
+
+Returns the matrix $M$ where all entries are replaced by their rounded values.
+
+$Example$
+==========
+julia> _round_matrix(QQ[1//3 -2 3//2; 5//3 0 -6; -7//3 -81//50 0])
+[0 -2 2]
+[2 0 -6] 
+[-2 -2 0]
+=#
+function _round_matrix(M::MatElem)
+  return map_entries(round, M)
+end
+
+
+#=
+  _abs_matrix(M::MatElem) -> MatElem
+
+Returns the matrix $M$ where all entries are replaced by their absolute values.
+
+$Example$
+==========
+julia> _abs_matrix(ZZ[1 -2 3; 4 0 -6; -7 -81 0])
+[1 2 3]
+[4 0 6] 
+[7 81 0]
+=#
+function _abs_matrix(M::MatElem)
+    return map_entries(abs, M)
+end
+
+#=
+  vecextract(M::MatElem,x::Int64,y::Int64) -> MatElem
+
+Extracts components of $M$ with regards to the binary representation of $x$ and $y$. 
+The location of the one entries give the position of the entries which should be extracted.
+For example if $x$ = $y$ = 13 = (1101)_2 then M[[1,3,4], [1,3,4]] is returned.
+
```

Comment:
When you write a doc, and this stands for all the others you have written, verbs are at the infinitive/imperative, so we do not use the 's' at the 3rd person. nb: for a matrix we talk about 'entries', not 'components.